result.js

```diff
@@ -24,25 +24,35 @@ const addAllSongsToPlaylist = response => {
   var playlistId = response.playlistId;
   var userId = response.userId;
   var songs = response.songs;
-  return new Promise(resolve => {
-    post("https://api.spotify.com/v1/users/" + userId + "/playlists/" + playlistId + "/tracks", {
-      Authorization: tokenType + ' ' + accessToken,
-      "Content-type": "application/json"
-    }, JSON.stringify({
-      uris: songs
-    }), function(response) {
-      progress.innerHTML += "[INFO] All songs has been added to the playlist...<br>";
-      resolve(response);
-    });
-  });
+  var songRequests = [];
+  // The spotify api only allows 100 songs to be sent across at a time
+  // We need to batch our requests.
+  while (songs.length > 0) {
+    var reqSongs = (songs.length > 100) ? songs.splice(0,100) : songs.splice(0, songs.length);
+    songRequests.push(new Promise(resolve => {
+      post("https://api.spotify.com/v1/users/" + userId + "/playlists/" + playlistId + "/tracks", {
+        Authorization: tokenType + ' ' + accessToken,
+        "Content-type": "application/json"
+      }, JSON.stringify({
+        uris: reqSongs
+      }), function(response) {
+        progress.innerHTML += "[INFO] " + reqSongs.length + " Songs have been added to the playlist...<br>";
+        resolve(response);
+      });
+    }));
+  }
+  return Promise.all(songRequests);
 };
 
 const prepareToaddAllSongsToPlaylist = response => {
   var songs = [];
   for (key in response) {
     if (isNumeric(key)) {
       try {
-        songs.push(response[key].uri);
+        // account for odd cases
+        if (response[key] && response[key].uri) {
```

Comment:
Thanks A LOT for adding this check as this should fix this problem, https://github.com/poanchen/google-play-music-playlist-exporter/issues/4 (and undefined error would not occur anymore) However, my only concern here is that (to the best of my knowledge), only the song which did not get any search result from Spotify API will happen to not have uri field and we simply just ignore them and print the error message to the console log (I will look into it a bit more to see if we can actually do something here but this can be tricky sometimes because the naming convention can be different for some songs).

Reply:
Perhaps we could add it to progress div, but I think we would have to do it in the searchASong function... this should work, but I haven't tested it yet:

```
 if ((!responseFromSearch.tracks.items[0] || (!responseFromSearch.tracks.items[0].uri)) {
   progress.innerHTML += '[INFO] Could not find song on Spotify: ' + response.song.title + ' by ' + response.song.artist + ' <br>';
}
```

Reply:
That looks like a good idea. I definitely prefer adding the warning into the progress div instead of the console log. I will work on adding that snippet of code to it and see if it works

Reply:
this is now related to #4

Reply:
just committed my fix (0475ac074132fc21f03ae0bc1100063390a5192d), now when the song could not be found in the Spotify search. It should show up in the progress div

Reply:
Great job! 

Thanks for the work you have done and for writing this plug-in. Writing it as a chrome extension was a great idea.

Reply:
Thanks @dfrencham for saying that! I greatly appreciate your contributions as well. Enjoy the rest of your day!